lib/msf/core/exploit/remote/smb/server/http_relay.rb

```diff
@@ -0,0 +1,162 @@
+###
+#
+# This mixin provides support for relaying smb logins to http servers
+#
+###
+
+module Msf::Exploit::Remote::SMB::Server::HttpRelay
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::NTLM::Client
+  include Msf::Exploit::Remote::SMB::Server::HashCapture
+
+  def post_login_action(client_socket, opts = {})
+    # overload for behavior after login
+  end
+
+  def http_client
+    return @http_client unless @http_client.nil?
+
+    @http_client = Rex::Proto::Http::Client.new(datastore['RHOST'],
+                                                datastore['RPORT'],
+                                                {},
+                                                datastore['SSL'],
+                                                datastore['ssl_version'],
+                                                proxies,
+                                                nil,
+                                                nil)
+    @http_client.connect
+    return @http_client
+  rescue Exception => e
+    print_error(e.to_s)
+    elog("Error connecting to Relay Target:\n #{e}")
+    raise e
+    return nil
+  end
+
+  def initial_handshake
+    vprint_status('Sending NTLM Handshake to HTTP Relay Target')
+    req = http_client.request_raw(
+      {
+        'method' => 'GET',
+        'uri' => @login_uri,
```

Comment:
Change to use the datastore URI

Reply:
On second thought, I like that we're using the instance variable created on the module side in the `on_client_connect`